suite-native/device-manager/src/components/DeviceManagerModal.tsx

```diff
@@ -1,47 +1,63 @@
 import { GestureResponderEvent, Modal, Pressable } from 'react-native';
 import { ReactNode } from 'react';
 import { useSafeAreaInsets, EdgeInsets } from 'react-native-safe-area-context';
-import Animated, { SlideInUp } from 'react-native-reanimated';
+import Animated, { FadeIn, SlideInUp } from 'react-native-reanimated';
+import { useSelector } from 'react-redux';
 
-import { ScreenHeaderWrapper, VStack } from '@suite-native/atoms';
+import { ScreenHeaderWrapper, Box, HStack } from '@suite-native/atoms';
 import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { selectDevice } from '@suite-common/wallet-core';
+import { AcquiredDevice } from '@suite-common/suite-types';
 
-import { DeviceSwitch } from './DeviceSwitch';
 import { useDeviceManager } from '../hooks/useDeviceManager';
+import { DeviceItemContent } from './DeviceItemContent';
 
 type DeviceManagerModalProps = {
     children: ReactNode;
+    customSwitchRighView?: ReactNode;
+    onClose?: () => void;
 };
 
-const MANAGER_MODAL_BOTTOM_RADIUS = 20;
+export const MANAGER_MODAL_BOTTOM_RADIUS = 12;
 
 const modalBackgroundOverlayStyle = prepareNativeStyle(utils => ({
     flex: 1,
     backgroundColor: utils.transparentize(0.3, utils.colors.backgroundNeutralBold),
 }));
 
-const deviceManagerModalWrapperStyle = prepareNativeStyle<{ insets: EdgeInsets }>(
+const deviceManagerModalWrapperStyle = prepareNativeStyle(utils => ({
+    backgroundColor: utils.colors.backgroundSurfaceElevation0,
+    borderBottomLeftRadius: MANAGER_MODAL_BOTTOM_RADIUS,
+    borderBottomRightRadius: MANAGER_MODAL_BOTTOM_RADIUS,
+}));
+
+const deviceSwitchWrapperStyle = prepareNativeStyle<{ insets: EdgeInsets }>(
     (utils, { insets }) => ({
-        paddingTop: Math.max(insets.top, utils.spacings.small),
+        paddingTop: insets.top + utils.spacings.large,
         backgroundColor: utils.colors.backgroundSurfaceElevation1,
         borderBottomLeftRadius: MANAGER_MODAL_BOTTOM_RADIUS,
         borderBottomRightRadius: MANAGER_MODAL_BOTTOM_RADIUS,
+        borderWidth: utils.borders.widths.small,
+        borderColor: utils.colors.borderElevation0,
+        zIndex: 20,
```

Comment:
Why is there zIndex here and on the animation as well?

Reply:
So things slide one under the another. Devices should slide under the rounded borders of header and wallets should slide under the devices with rounded borders
![Uploading Simulator Screenshot - iPhone 15 - 2024-05-07 at 22.15.58.png…]()